### PR TITLE
feature: add top donors endpoint

### LINKED ISSUE #4415: feature: add top donors report endpoint
# Feature Request

## User Story
As a developer, I want a top donors Reports API endpoint so that a Top Donors list can be rendered on the Reports page for a given period.

## Possible Solution
Use existing data to gather top donors for a given period, and return an array of objects representing list items in the appropriate shape for the RESTList component.

## Acceptance Criteria
- [x] Does not re-invent the wheel, uses existing logic to aggregate top donor data
- [x] Returns data in the shape appropriate for the RESTList component to render DonorItems
- [x] Returns data depending on the provided period

### PR DESCRIPTION
**Pull request by henryholtgeerts**
## Description
Resolves #4415 
This PR creates a new 'top-donors' endpoint which uses the existing Give_Donors_Query class to aggregate top donor information for a given period. The endpoint returns the data in the correct shape for use in the RESTList component. The PR also address some issues with the RESTList component, and adds a new URL prop to the DonorItem component, making an item linking through to the appropriate donor overview page.

## How Has This Been Tested?
It has been tested locally and throws no errors in the browser console. Passed PHPUnit tests.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: assets/src/js/admin/reports/components/donor-item/index.js [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
 
 DonorItem.defaultProps = {
-    image: null,
-    name: 'Anonymous',
-    email: null,
-    count: null,
-    total: null
-}
+	image: null,
+	name: 'Anonymous',
```
Comment (ravinderk) [inline assets/src/js/admin/reports/components/donor-item/index.js]: @henryholtgeerts Can we localize this string?

File: src/API/Endpoints/Reports/TopDonors.php [hunk 2]
```diff
@@ -14,21 +14,48 @@ public function __construct() {
...
+			$item = [
+				'type'  => 'donor',
+				'name'  => $donor->name,
+				'url'   => $url,
+				'count' => "{$donor->purchase_count} {$countLabel}",
+				'total' => $total,
+				'image' => $avatar,
+				'email' => $donor->email,
+			];
+			array_push( $list, $item );
```
Comment (ravinderk) [inline src/API/Endpoints/Reports/TopDonors.php]: @henryholtgeerts can we use the shortened method.
I suggested same here: https://github.com/impress-org/givewp/pull/4420#discussion_r368825393

File: assets/src/js/admin/reports/components/donor-item/index.js (hunk 1 above)
Comment (henryholtgeerts) [reply assets/src/js/admin/reports/components/donor-item/index.js]: Change made, ref: https://github.com/impress-org/givewp/pull/4430/commits/644b10b42bbe6b721aa38bda5b6470a32553d763

File: src/API/Endpoints/Reports/TopDonors.php (hunk 2 above)
Comment (henryholtgeerts) [reply src/API/Endpoints/Reports/TopDonors.php]: Change made, ref: https://github.com/impress-org/givewp/pull/4430/commits/da20a7bd102ef43e220108b9f93effe1933c4068

**Pull request merged**